DigitalLearningSolutions.Data.Tests/Services/UserServiceTests.cs

```diff
@@ -1,66 +1,106 @@
 ﻿namespace DigitalLearningSolutions.Data.Tests.Services
 {
+    using System.Collections.Generic;
     using System.Linq;
+    using DigitalLearningSolutions.Data.Models.User;
     using DigitalLearningSolutions.Data.Services;
     using DigitalLearningSolutions.Data.Tests.Helpers;
+    using FakeItEasy;
     using FluentAssertions;
     using NUnit.Framework;
 
     public class UserServiceTests
     {
         private IUserService userService;
+        private IUserDataService userDataService;
 
         [SetUp]
         public void Setup()
         {
-            var connection = ServiceTestHelper.GetDatabaseConnection();
-            userService = new UserService(connection);
+            userDataService = A.Fake<IUserDataService>();
+            userService = new UserService(userDataService);
         }
 
         [Test]
-        public void GetAdminUserByUsername_Returns_admin_user()
+        public void GetUsersByUsername_Returns_admin_user_and_delegate_users()
         {
             // Given
             var expectedAdminUser = UserTestHelper.GetDefaultAdminUser();
+            var expectedDelegateUsers = UserTestHelper.GetDefaultDelegateUser();
+            A.CallTo(() => userDataService.GetAdminUserByUsername(A<string>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            A.CallTo(() => userDataService.GetDelegateUsersByUsername(A<string>._)).Returns(new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() });
 
             //When
-            var returnedAdminUser = userService.GetAdminUserByUsername("Username");
+            var (returnedAdminUser, _) = userService.GetUsersByUsername("Username");
+            var (_, returnedDelegateUsers) = userService.GetUsersByUsername("SV1234");
```

Comment:
Given that we need to call this method twice, with different usernames, could we split it into two different tests?

Reply:
Unsure, I didn't write the initial unit test, The only change I made to it was because I added the userDataService dependency for the methods it calls. I can look to change it

Reply:
I think I initially wrote this test and it was just modified as part of the changes here. While it was originally written to test the scenario of a function returning both in one call, because of the way the test users are set up, they cannot both be returned at once. So I suppose splitting it into two might be best.

Reply:
Since this test is no longer testing against the database, it doesn't actually matter what string we are inputting since the fake UserDataService is accepting A<string>._. Combined it into a single call in a single test. There are separate tests in the UserDataServiceTests that test against the actual data